### PR TITLE
Allow different branch name format

### PR DESCRIPTION
**Pull request by florianmski**
### Description
One of the main issue that blocked me from using this daily at work was that we are using a different branch naming pattern than the hardcoded one in `gong`.  
Namely something of this format: `XXX-ID/TITLE` with illegal characters being replaced by a `_` rather than the hardcoded `-`.  
Did my best to introduce this, please do comment on the style if something doesn't look right, also I have no experience whatsoever with Go so I might be doing things the wrong way 😅

### Implementation Considerations
- Introduce two new config parameters: `branch_replacement_character ` and `branch_pattern `, see the also modified `Readme.md` for more infos
- Extracted configuration logic into its own `configuration.go` file, no code changed there
- Introduce a `branch.go` file to handle the branch name generation according to the config parameters
- Some more bits and pieces, see comments in the PR

### Testing
- Added a test to the `slugiffy.go` class to handle a edge case I found
- Added tests to the `branch.go` class

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -122,56 +118,3 @@ func Login(client Client) (bool, error) {
...
-	}
-
-	err = ioutil.WriteFile(fileLocation, loginDetails, 0644)
-
-	if err != nil {
-		return err
-	}
-
-	return nil
-}
```
Comment (florianmski) [inline client.go:177]: Moved all of this to `configuration.go`, didn't modify any of this code

File: slugger.go [hunk 2]
```diff
@@ -6,8 +6,9 @@ import (
...
 // SlugifyTitle : Make sure to clean up the branch names
-func SlugifyTitle(ticketTitle string) string {
-	re := regexp.MustCompile("[^a-z0-9]+")
-
-	return strings.Trim(re.ReplaceAllString(strings.ToLower(ticketTitle), "-"), "-")
+func SlugifyTitle(ticketTitle string, replacementCharacter string) string {
+	re := regexp.MustCompile("[^a-z0-9]+");
+	lowerCaseTitle := strings.ToLower(ticketTitle)
+	formattedTitle := re.ReplaceAllString(lowerCaseTitle, replacementCharacter)
+	return strings.Trim(formattedTitle, replacementCharacter)
```
Comment (florianmski) [inline slugger.go:13]: Was a bit puzzled on how this was working, tried to make it a bit clearer by adding some more steps

File: slugger_test.go [hunk 3]
```diff
@@ -13,15 +13,20 @@ var _ = Suite(&SluggerSuite{})
...
 func (s *SluggerSuite) TestSlugCreationWithMultipleSpaces(c *C) {
 	title := "This is a                    fake title"
-	c.Assert(SlugifyTitle(title), Equals, "this-is-a-fake-title")
+	c.Assert(SlugifyTitle(title, "+"), Equals, "this+is+a+fake+title")
+}
+
+func (s *SluggerSuite) TestSlugCreationWithSpecialCharAtTheEnd(c *C) {
+	title := "This is a fake (title)"
+	c.Assert(SlugifyTitle(title, "+"), Equals, "this+is+a+fake+title")
 }
```
Comment (florianmski) [inline slugger_test.go:32]: Added this test as my modifications to the logic of the `SlugifyTitle` class introduced a bug in some edge case
